src/adapt.rs

```diff
@@ -0,0 +1,481 @@
+//! An API for plugging in adapters.
```

Comment:
Well, there is already a module `adapters`. I could move this in `adapters/mod.rs`, but that would make that file too large for my tastes. I could also move this to, say, `adapters/api.rs`, if you prefer.